From df55953552b4fea12d789a44fd1e7da36937b8c4 Mon Sep 17 00:00:00 2001
From: Jesus Camacho Rodriguez <jcamacho@apache.org>
Date: Sat, 1 Oct 2016 09:30:35 +0100
Subject: [PATCH 0879/1164] CDH-45809: HIVE-14873: Add UDF for extraction of
 'day of week' (Jesus Camacho Rodriguez, reviewed
 by Ashutosh Chauhan)

(cherry picked from commit 59539885725a96cca4b3f0759a5b26e0d8198dc8)

Change-Id: I554fabb1b3edf61884b3d26e3577fa2cdf5f8bfa
---
 .../hadoop/hive/ql/exec/FunctionRegistry.java      |    2 +
 .../apache/hadoop/hive/ql/udf/UDFDayOfWeek.java    |   98 +++
 .../queries/clientpositive/vectorized_date_funcs.q |    8 +-
 .../results/clientpositive/show_functions.q.out    |    1 +
 .../clientpositive/vectorized_date_funcs.q.out     |  849 ++++++++++----------
 5 files changed, 537 insertions(+), 421 deletions(-)
 create mode 100644 ql/src/java/org/apache/hadoop/hive/ql/udf/UDFDayOfWeek.java

diff --git a/ql/src/java/org/apache/hadoop/hive/ql/exec/FunctionRegistry.java b/ql/src/java/org/apache/hadoop/hive/ql/exec/FunctionRegistry.java
index 931f88d..6812f5c 100644
--- a/ql/src/java/org/apache/hadoop/hive/ql/exec/FunctionRegistry.java
+++ b/ql/src/java/org/apache/hadoop/hive/ql/exec/FunctionRegistry.java
@@ -52,6 +52,7 @@
 import org.apache.hadoop.hive.ql.udf.UDFCos;
 import org.apache.hadoop.hive.ql.udf.UDFCrc32;
 import org.apache.hadoop.hive.ql.udf.UDFDayOfMonth;
+import org.apache.hadoop.hive.ql.udf.UDFDayOfWeek;
 import org.apache.hadoop.hive.ql.udf.UDFDegrees;
 import org.apache.hadoop.hive.ql.udf.UDFE;
 import org.apache.hadoop.hive.ql.udf.UDFExp;
@@ -261,6 +262,7 @@
 
     system.registerUDF("day", UDFDayOfMonth.class, false);
     system.registerUDF("dayofmonth", UDFDayOfMonth.class, false);
+    system.registerUDF("dayofweek", UDFDayOfWeek.class, false);
     system.registerUDF("month", UDFMonth.class, false);
     system.registerUDF("year", UDFYear.class, false);
     system.registerUDF("hour", UDFHour.class, false);
diff --git a/ql/src/java/org/apache/hadoop/hive/ql/udf/UDFDayOfWeek.java b/ql/src/java/org/apache/hadoop/hive/ql/udf/UDFDayOfWeek.java
new file mode 100644
index 0000000..2b16108
--- /dev/null
+++ b/ql/src/java/org/apache/hadoop/hive/ql/udf/UDFDayOfWeek.java
@@ -0,0 +1,98 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.hive.ql.udf;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+
+import org.apache.hadoop.hive.ql.exec.Description;
+import org.apache.hadoop.hive.ql.exec.UDF;
+import org.apache.hadoop.hive.serde2.io.DateWritable;
+import org.apache.hadoop.hive.serde2.io.TimestampWritable;
+import org.apache.hadoop.io.IntWritable;
+import org.apache.hadoop.io.Text;
+
+/**
+ * UDFDayOfWeek.
+ *
+ */
+@Description(name = "dayofweek",
+    value = "_FUNC_(param) - Returns the day of the week of date/timestamp "
+    + "(1 = Sunday, 2 = Monday, ..., 7 = Saturday)",
+    extended = "param can be one of:\n"
+    + "1. A string in the format of 'yyyy-MM-dd HH:mm:ss' or 'yyyy-MM-dd'.\n"
+    + "2. A date value\n"
+    + "3. A timestamp value"
+    + "Example:\n "
+    + "  > SELECT _FUNC_('2009-07-30') FROM src LIMIT 1;\n" + "  5")
+public class UDFDayOfWeek extends UDF {
+  private final SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd");
+  private final Calendar calendar = Calendar.getInstance();
+
+  private final IntWritable result = new IntWritable();
+
+  public UDFDayOfWeek() {
+  }
+
+  /**
+   * Get the day of week from a date string.
+   *
+   * @param dateString
+   *          the dateString in the format of "yyyy-MM-dd HH:mm:ss" or
+   *          "yyyy-MM-dd".
+   * @return an int from 1 to 7. null if the dateString is not a valid date
+   *         string.
+   */
+  public IntWritable evaluate(Text dateString) {
+    if (dateString == null) {
+      return null;
+    }
+    try {
+      Date date = formatter.parse(dateString.toString());
+      calendar.setTime(date);
+      result.set(calendar.get(Calendar.DAY_OF_WEEK));
+      return result;
+    } catch (ParseException e) {
+      return null;
+    }
+  }
+
+  public IntWritable evaluate(DateWritable d) {
+    if (d == null) {
+      return null;
+    }
+
+    calendar.setTime(d.get()); // Time doesn't matter.
+    result.set(calendar.get(Calendar.DAY_OF_WEEK));
+    return result;
+  }
+
+  public IntWritable evaluate(TimestampWritable t) {
+    if (t == null) {
+      return null;
+    }
+
+    calendar.setTime(t.getTimestamp());
+    result.set(calendar.get(Calendar.DAY_OF_WEEK));
+    return result;
+  }
+
+}
diff --git a/ql/src/test/queries/clientpositive/vectorized_date_funcs.q b/ql/src/test/queries/clientpositive/vectorized_date_funcs.q
index 1fb0dac..46ef0dc 100644
--- a/ql/src/test/queries/clientpositive/vectorized_date_funcs.q
+++ b/ql/src/test/queries/clientpositive/vectorized_date_funcs.q
@@ -26,6 +26,7 @@ EXPLAIN SELECT
   month(fl_time),
   day(fl_time),
   dayofmonth(fl_time),
+  dayofweek(fl_time),
   weekofyear(fl_time),
   date(fl_time),
   to_date(fl_time),
@@ -40,6 +41,7 @@ SELECT
   month(fl_time),
   day(fl_time),
   dayofmonth(fl_time),
+  dayofweek(fl_time),
   weekofyear(fl_time),
   date(fl_time),
   to_date(fl_time),
@@ -54,6 +56,7 @@ EXPLAIN SELECT
   month(fl_date),
   day(fl_date),
   dayofmonth(fl_date),
+  dayofweek(fl_date),
   weekofyear(fl_date),
   date(fl_date),
   to_date(fl_date),
@@ -68,6 +71,7 @@ SELECT
   month(fl_date),
   day(fl_date),
   dayofmonth(fl_date),
+  dayofweek(fl_date),
   weekofyear(fl_date),
   date(fl_date),
   to_date(fl_date),
@@ -81,6 +85,7 @@ EXPLAIN SELECT
   month(fl_time) = month(fl_date),
   day(fl_time) = day(fl_date),
   dayofmonth(fl_time) = dayofmonth(fl_date),
+  dayofweek(fl_time) = dayofweek(fl_date),
   weekofyear(fl_time) = weekofyear(fl_date),
   date(fl_time) = date(fl_date),
   to_date(fl_time) = to_date(fl_date),
@@ -95,6 +100,7 @@ SELECT
   month(fl_time) = month(fl_date),
   day(fl_time) = day(fl_date),
   dayofmonth(fl_time) = dayofmonth(fl_date),
+  dayofweek(fl_time) = dayofweek(fl_date),
   weekofyear(fl_time) = weekofyear(fl_date),
   date(fl_time) = date(fl_date),
   to_date(fl_time) = to_date(fl_date),
@@ -138,4 +144,4 @@ SELECT
   count(fl_date),
   count(*)
 FROM date_udf_flight_orc
-ORDER BY c1;
\ No newline at end of file
+ORDER BY c1;
diff --git a/ql/src/test/results/clientpositive/show_functions.q.out b/ql/src/test/results/clientpositive/show_functions.q.out
index 52b1424..794e9cb 100644
--- a/ql/src/test/results/clientpositive/show_functions.q.out
+++ b/ql/src/test/results/clientpositive/show_functions.q.out
@@ -63,6 +63,7 @@ date_sub
 datediff
 day
 dayofmonth
+dayofweek
 decode
 default.qtest_get_java_boolean
 degrees
diff --git a/ql/src/test/results/clientpositive/vectorized_date_funcs.q.out b/ql/src/test/results/clientpositive/vectorized_date_funcs.q.out
index 8cde444..443e9e4 100644
--- a/ql/src/test/results/clientpositive/vectorized_date_funcs.q.out
+++ b/ql/src/test/results/clientpositive/vectorized_date_funcs.q.out
@@ -205,6 +205,7 @@ PREHOOK: query: EXPLAIN SELECT
   month(fl_time),
   day(fl_time),
   dayofmonth(fl_time),
+  dayofweek(fl_time),
   weekofyear(fl_time),
   date(fl_time),
   to_date(fl_time),
@@ -219,6 +220,7 @@ POSTHOOK: query: EXPLAIN SELECT
   month(fl_time),
   day(fl_time),
   dayofmonth(fl_time),
+  dayofweek(fl_time),
   weekofyear(fl_time),
   date(fl_time),
   to_date(fl_time),
@@ -239,8 +241,8 @@ STAGE PLANS:
             alias: date_udf_flight_orc
             Statistics: Num rows: 137 Data size: 13152 Basic stats: COMPLETE Column stats: NONE
             Select Operator
-              expressions: to_unix_timestamp(fl_time) (type: bigint), year(fl_time) (type: int), month(fl_time) (type: int), day(fl_time) (type: int), dayofmonth(fl_time) (type: int), weekofyear(fl_time) (type: int), CAST( fl_time AS DATE) (type: date), to_date(fl_time) (type: string), date_add(fl_time, 2) (type: string), date_sub(fl_time, 2) (type: string), datediff(fl_time, '2000-01-01') (type: int)
-              outputColumnNames: _col0, _col1, _col2, _col3, _col4, _col5, _col6, _col7, _col8, _col9, _col10
+              expressions: to_unix_timestamp(fl_time) (type: bigint), year(fl_time) (type: int), month(fl_time) (type: int), day(fl_time) (type: int), dayofmonth(fl_time) (type: int), dayofweek(fl_time) (type: int), weekofyear(fl_time) (type: int), CAST( fl_time AS DATE) (type: date), to_date(fl_time) (type: string), date_add(fl_time, 2) (type: string), date_sub(fl_time, 2) (type: string), datediff(fl_time, '2000-01-01') (type: int)
+              outputColumnNames: _col0, _col1, _col2, _col3, _col4, _col5, _col6, _col7, _col8, _col9, _col10, _col11
               Statistics: Num rows: 137 Data size: 13152 Basic stats: COMPLETE Column stats: NONE
               File Output Operator
                 compressed: false
@@ -249,7 +251,6 @@ STAGE PLANS:
                     input format: org.apache.hadoop.mapred.TextInputFormat
                     output format: org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat
                     serde: org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe
-      Execution mode: vectorized
 
   Stage: Stage-0
     Fetch Operator
@@ -263,6 +264,7 @@ PREHOOK: query: SELECT
   month(fl_time),
   day(fl_time),
   dayofmonth(fl_time),
+  dayofweek(fl_time),
   weekofyear(fl_time),
   date(fl_time),
   to_date(fl_time),
@@ -279,6 +281,7 @@ POSTHOOK: query: SELECT
   month(fl_time),
   day(fl_time),
   dayofmonth(fl_time),
+  dayofweek(fl_time),
   weekofyear(fl_time),
   date(fl_time),
   to_date(fl_time),
@@ -289,149 +292,150 @@ FROM date_udf_flight_orc
 POSTHOOK: type: QUERY
 POSTHOOK: Input: default@date_udf_flight_orc
 #### A masked pattern was here ####
-1287583200	2010	10	20	20	42	2010-10-20	2010-10-20	2010-10-22	2010-10-18	3945
-1287583200	2010	10	20	20	42	2010-10-20	2010-10-20	2010-10-22	2010-10-18	3945
-1287583200	2010	10	20	20	42	2010-10-20	2010-10-20	2010-10-22	2010-10-18	3945
-1287583200	2010	10	20	20	42	2010-10-20	2010-10-20	2010-10-22	2010-10-18	3945
-1287583200	2010	10	20	20	42	2010-10-20	2010-10-20	2010-10-22	2010-10-18	3945
-1287583200	2010	10	20	20	42	2010-10-20	2010-10-20	2010-10-22	2010-10-18	3945
-1287583200	2010	10	20	20	42	2010-10-20	2010-10-20	2010-10-22	2010-10-18	3945
-1287583200	2010	10	20	20	42	2010-10-20	2010-10-20	2010-10-22	2010-10-18	3945
-1287669600	2010	10	21	21	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
-1287669600	2010	10	21	21	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
-1287669600	2010	10	21	21	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
-1287669600	2010	10	21	21	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
-1287669600	2010	10	21	21	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
-1287669600	2010	10	21	21	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
-1287669600	2010	10	21	21	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
-1287669600	2010	10	21	21	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
-1287756000	2010	10	22	22	42	2010-10-22	2010-10-22	2010-10-24	2010-10-20	3947
-1287756000	2010	10	22	22	42	2010-10-22	2010-10-22	2010-10-24	2010-10-20	3947
-1287756000	2010	10	22	22	42	2010-10-22	2010-10-22	2010-10-24	2010-10-20	3947
-1287756000	2010	10	22	22	42	2010-10-22	2010-10-22	2010-10-24	2010-10-20	3947
-1287756000	2010	10	22	22	42	2010-10-22	2010-10-22	2010-10-24	2010-10-20	3947
-1287756000	2010	10	22	22	42	2010-10-22	2010-10-22	2010-10-24	2010-10-20	3947
-1287756000	2010	10	22	22	42	2010-10-22	2010-10-22	2010-10-24	2010-10-20	3947
-1287756000	2010	10	22	22	42	2010-10-22	2010-10-22	2010-10-24	2010-10-20	3947
-1287842400	2010	10	23	23	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
-1287842400	2010	10	23	23	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
-1287842400	2010	10	23	23	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
-1287842400	2010	10	23	23	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
-1287842400	2010	10	23	23	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
-1287842400	2010	10	23	23	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
-1287842400	2010	10	23	23	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
-1287928800	2010	10	24	24	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
-1287928800	2010	10	24	24	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
-1287928800	2010	10	24	24	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
-1287928800	2010	10	24	24	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
-1287928800	2010	10	24	24	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
-1287928800	2010	10	24	24	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
-1287928800	2010	10	24	24	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
-1288015200	2010	10	25	25	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
-1288015200	2010	10	25	25	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
-1288015200	2010	10	25	25	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
-1288015200	2010	10	25	25	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
-1288015200	2010	10	25	25	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
-1288015200	2010	10	25	25	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
-1288015200	2010	10	25	25	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
-1288015200	2010	10	25	25	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
-1288101600	2010	10	26	26	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
-1288101600	2010	10	26	26	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
-1288101600	2010	10	26	26	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
-1288101600	2010	10	26	26	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
-1288101600	2010	10	26	26	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
-1288101600	2010	10	26	26	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
-1288101600	2010	10	26	26	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
-1288101600	2010	10	26	26	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
-1288188000	2010	10	27	27	43	2010-10-27	2010-10-27	2010-10-29	2010-10-25	3952
-1288188000	2010	10	27	27	43	2010-10-27	2010-10-27	2010-10-29	2010-10-25	3952
-1288188000	2010	10	27	27	43	2010-10-27	2010-10-27	2010-10-29	2010-10-25	3952
-1288188000	2010	10	27	27	43	2010-10-27	2010-10-27	2010-10-29	2010-10-25	3952
-1288188000	2010	10	27	27	43	2010-10-27	2010-10-27	2010-10-29	2010-10-25	3952
-1288188000	2010	10	27	27	43	2010-10-27	2010-10-27	2010-10-29	2010-10-25	3952
-1288188000	2010	10	27	27	43	2010-10-27	2010-10-27	2010-10-29	2010-10-25	3952
-1288188000	2010	10	27	27	43	2010-10-27	2010-10-27	2010-10-29	2010-10-25	3952
-1288274400	2010	10	28	28	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
-1288274400	2010	10	28	28	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
-1288274400	2010	10	28	28	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
-1288274400	2010	10	28	28	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
-1288274400	2010	10	28	28	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
-1288274400	2010	10	28	28	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
-1288274400	2010	10	28	28	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
-1288274400	2010	10	28	28	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
-1288360800	2010	10	29	29	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
-1288360800	2010	10	29	29	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
-1288360800	2010	10	29	29	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
-1288360800	2010	10	29	29	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
-1288360800	2010	10	29	29	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
-1288360800	2010	10	29	29	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
-1288360800	2010	10	29	29	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
-1288360800	2010	10	29	29	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
-1288447200	2010	10	30	30	43	2010-10-30	2010-10-30	2010-11-01	2010-10-28	3955
-1288447200	2010	10	30	30	43	2010-10-30	2010-10-30	2010-11-01	2010-10-28	3955
-1288447200	2010	10	30	30	43	2010-10-30	2010-10-30	2010-11-01	2010-10-28	3955
-1288447200	2010	10	30	30	43	2010-10-30	2010-10-30	2010-11-01	2010-10-28	3955
-1288447200	2010	10	30	30	43	2010-10-30	2010-10-30	2010-11-01	2010-10-28	3955
-1288447200	2010	10	30	30	43	2010-10-30	2010-10-30	2010-11-01	2010-10-28	3955
-1288447200	2010	10	30	30	43	2010-10-30	2010-10-30	2010-11-01	2010-10-28	3955
-1288533600	2010	10	31	31	43	2010-10-31	2010-10-31	2010-11-02	2010-10-29	3956
-1288533600	2010	10	31	31	43	2010-10-31	2010-10-31	2010-11-02	2010-10-29	3956
-1288533600	2010	10	31	31	43	2010-10-31	2010-10-31	2010-11-02	2010-10-29	3956
-1288533600	2010	10	31	31	43	2010-10-31	2010-10-31	2010-11-02	2010-10-29	3956
-1288533600	2010	10	31	31	43	2010-10-31	2010-10-31	2010-11-02	2010-10-29	3956
-1288533600	2010	10	31	31	43	2010-10-31	2010-10-31	2010-11-02	2010-10-29	3956
-1288533600	2010	10	31	31	43	2010-10-31	2010-10-31	2010-11-02	2010-10-29	3956
-1288447200	2010	10	30	30	43	2010-10-30	2010-10-30	2010-11-01	2010-10-28	3955
-1288447200	2010	10	30	30	43	2010-10-30	2010-10-30	2010-11-01	2010-10-28	3955
-1288360800	2010	10	29	29	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
-1288360800	2010	10	29	29	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
-1288360800	2010	10	29	29	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
-1288274400	2010	10	28	28	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
-1288274400	2010	10	28	28	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
-1288274400	2010	10	28	28	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
-1288188000	2010	10	27	27	43	2010-10-27	2010-10-27	2010-10-29	2010-10-25	3952
-1288188000	2010	10	27	27	43	2010-10-27	2010-10-27	2010-10-29	2010-10-25	3952
-1288101600	2010	10	26	26	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
-1288101600	2010	10	26	26	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
-1288101600	2010	10	26	26	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
-1288101600	2010	10	26	26	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
-1288015200	2010	10	25	25	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
-1288015200	2010	10	25	25	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
-1288015200	2010	10	25	25	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
-1287928800	2010	10	24	24	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
-1287928800	2010	10	24	24	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
-1287928800	2010	10	24	24	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
-1287928800	2010	10	24	24	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
-1287842400	2010	10	23	23	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
-1287756000	2010	10	22	22	42	2010-10-22	2010-10-22	2010-10-24	2010-10-20	3947
-1287756000	2010	10	22	22	42	2010-10-22	2010-10-22	2010-10-24	2010-10-20	3947
-1287756000	2010	10	22	22	42	2010-10-22	2010-10-22	2010-10-24	2010-10-20	3947
-1287669600	2010	10	21	21	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
-1287669600	2010	10	21	21	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
-1287669600	2010	10	21	21	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
-1287583200	2010	10	20	20	42	2010-10-20	2010-10-20	2010-10-22	2010-10-18	3945
-1287583200	2010	10	20	20	42	2010-10-20	2010-10-20	2010-10-22	2010-10-18	3945
-1287842400	2010	10	23	23	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
-1287842400	2010	10	23	23	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
-1287842400	2010	10	23	23	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
-1288447200	2010	10	30	30	43	2010-10-30	2010-10-30	2010-11-01	2010-10-28	3955
-1288447200	2010	10	30	30	43	2010-10-30	2010-10-30	2010-11-01	2010-10-28	3955
-1287583200	2010	10	20	20	42	2010-10-20	2010-10-20	2010-10-22	2010-10-18	3945
-1287669600	2010	10	21	21	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
-1287842400	2010	10	23	23	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
-1287928800	2010	10	24	24	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
-1288015200	2010	10	25	25	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
-1288101600	2010	10	26	26	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
-1288188000	2010	10	27	27	43	2010-10-27	2010-10-27	2010-10-29	2010-10-25	3952
-1288274400	2010	10	28	28	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
-1288360800	2010	10	29	29	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
-1288533600	2010	10	31	31	43	2010-10-31	2010-10-31	2010-11-02	2010-10-29	3956
+1287583200	2010	10	20	20	4	42	2010-10-20	2010-10-20	2010-10-22	2010-10-18	3945
+1287583200	2010	10	20	20	4	42	2010-10-20	2010-10-20	2010-10-22	2010-10-18	3945
+1287583200	2010	10	20	20	4	42	2010-10-20	2010-10-20	2010-10-22	2010-10-18	3945
+1287583200	2010	10	20	20	4	42	2010-10-20	2010-10-20	2010-10-22	2010-10-18	3945
+1287583200	2010	10	20	20	4	42	2010-10-20	2010-10-20	2010-10-22	2010-10-18	3945
+1287583200	2010	10	20	20	4	42	2010-10-20	2010-10-20	2010-10-22	2010-10-18	3945
+1287583200	2010	10	20	20	4	42	2010-10-20	2010-10-20	2010-10-22	2010-10-18	3945
+1287583200	2010	10	20	20	4	42	2010-10-20	2010-10-20	2010-10-22	2010-10-18	3945
+1287669600	2010	10	21	21	5	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
+1287669600	2010	10	21	21	5	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
+1287669600	2010	10	21	21	5	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
+1287669600	2010	10	21	21	5	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
+1287669600	2010	10	21	21	5	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
+1287669600	2010	10	21	21	5	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
+1287669600	2010	10	21	21	5	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
+1287669600	2010	10	21	21	5	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
+1287756000	2010	10	22	22	6	42	2010-10-22	2010-10-22	2010-10-24	2010-10-20	3947
+1287756000	2010	10	22	22	6	42	2010-10-22	2010-10-22	2010-10-24	2010-10-20	3947
+1287756000	2010	10	22	22	6	42	2010-10-22	2010-10-22	2010-10-24	2010-10-20	3947
+1287756000	2010	10	22	22	6	42	2010-10-22	2010-10-22	2010-10-24	2010-10-20	3947
+1287756000	2010	10	22	22	6	42	2010-10-22	2010-10-22	2010-10-24	2010-10-20	3947
+1287756000	2010	10	22	22	6	42	2010-10-22	2010-10-22	2010-10-24	2010-10-20	3947
+1287756000	2010	10	22	22	6	42	2010-10-22	2010-10-22	2010-10-24	2010-10-20	3947
+1287756000	2010	10	22	22	6	42	2010-10-22	2010-10-22	2010-10-24	2010-10-20	3947
+1287842400	2010	10	23	23	7	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
+1287842400	2010	10	23	23	7	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
+1287842400	2010	10	23	23	7	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
+1287842400	2010	10	23	23	7	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
+1287842400	2010	10	23	23	7	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
+1287842400	2010	10	23	23	7	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
+1287842400	2010	10	23	23	7	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
+1287928800	2010	10	24	24	1	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
+1287928800	2010	10	24	24	1	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
+1287928800	2010	10	24	24	1	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
+1287928800	2010	10	24	24	1	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
+1287928800	2010	10	24	24	1	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
+1287928800	2010	10	24	24	1	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
+1287928800	2010	10	24	24	1	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
+1288015200	2010	10	25	25	2	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
+1288015200	2010	10	25	25	2	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
+1288015200	2010	10	25	25	2	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
+1288015200	2010	10	25	25	2	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
+1288015200	2010	10	25	25	2	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
+1288015200	2010	10	25	25	2	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
+1288015200	2010	10	25	25	2	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
+1288015200	2010	10	25	25	2	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
+1288101600	2010	10	26	26	3	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
+1288101600	2010	10	26	26	3	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
+1288101600	2010	10	26	26	3	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
+1288101600	2010	10	26	26	3	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
+1288101600	2010	10	26	26	3	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
+1288101600	2010	10	26	26	3	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
+1288101600	2010	10	26	26	3	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
+1288101600	2010	10	26	26	3	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
+1288188000	2010	10	27	27	4	43	2010-10-27	2010-10-27	2010-10-29	2010-10-25	3952
+1288188000	2010	10	27	27	4	43	2010-10-27	2010-10-27	2010-10-29	2010-10-25	3952
+1288188000	2010	10	27	27	4	43	2010-10-27	2010-10-27	2010-10-29	2010-10-25	3952
+1288188000	2010	10	27	27	4	43	2010-10-27	2010-10-27	2010-10-29	2010-10-25	3952
+1288188000	2010	10	27	27	4	43	2010-10-27	2010-10-27	2010-10-29	2010-10-25	3952
+1288188000	2010	10	27	27	4	43	2010-10-27	2010-10-27	2010-10-29	2010-10-25	3952
+1288188000	2010	10	27	27	4	43	2010-10-27	2010-10-27	2010-10-29	2010-10-25	3952
+1288188000	2010	10	27	27	4	43	2010-10-27	2010-10-27	2010-10-29	2010-10-25	3952
+1288274400	2010	10	28	28	5	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
+1288274400	2010	10	28	28	5	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
+1288274400	2010	10	28	28	5	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
+1288274400	2010	10	28	28	5	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
+1288274400	2010	10	28	28	5	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
+1288274400	2010	10	28	28	5	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
+1288274400	2010	10	28	28	5	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
+1288274400	2010	10	28	28	5	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
+1288360800	2010	10	29	29	6	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
+1288360800	2010	10	29	29	6	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
+1288360800	2010	10	29	29	6	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
+1288360800	2010	10	29	29	6	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
+1288360800	2010	10	29	29	6	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
+1288360800	2010	10	29	29	6	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
+1288360800	2010	10	29	29	6	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
+1288360800	2010	10	29	29	6	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
+1288447200	2010	10	30	30	7	43	2010-10-30	2010-10-30	2010-11-01	2010-10-28	3955
+1288447200	2010	10	30	30	7	43	2010-10-30	2010-10-30	2010-11-01	2010-10-28	3955
+1288447200	2010	10	30	30	7	43	2010-10-30	2010-10-30	2010-11-01	2010-10-28	3955
+1288447200	2010	10	30	30	7	43	2010-10-30	2010-10-30	2010-11-01	2010-10-28	3955
+1288447200	2010	10	30	30	7	43	2010-10-30	2010-10-30	2010-11-01	2010-10-28	3955
+1288447200	2010	10	30	30	7	43	2010-10-30	2010-10-30	2010-11-01	2010-10-28	3955
+1288447200	2010	10	30	30	7	43	2010-10-30	2010-10-30	2010-11-01	2010-10-28	3955
+1288533600	2010	10	31	31	1	43	2010-10-31	2010-10-31	2010-11-02	2010-10-29	3956
+1288533600	2010	10	31	31	1	43	2010-10-31	2010-10-31	2010-11-02	2010-10-29	3956
+1288533600	2010	10	31	31	1	43	2010-10-31	2010-10-31	2010-11-02	2010-10-29	3956
+1288533600	2010	10	31	31	1	43	2010-10-31	2010-10-31	2010-11-02	2010-10-29	3956
+1288533600	2010	10	31	31	1	43	2010-10-31	2010-10-31	2010-11-02	2010-10-29	3956
+1288533600	2010	10	31	31	1	43	2010-10-31	2010-10-31	2010-11-02	2010-10-29	3956
+1288533600	2010	10	31	31	1	43	2010-10-31	2010-10-31	2010-11-02	2010-10-29	3956
+1288447200	2010	10	30	30	7	43	2010-10-30	2010-10-30	2010-11-01	2010-10-28	3955
+1288447200	2010	10	30	30	7	43	2010-10-30	2010-10-30	2010-11-01	2010-10-28	3955
+1288360800	2010	10	29	29	6	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
+1288360800	2010	10	29	29	6	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
+1288360800	2010	10	29	29	6	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
+1288274400	2010	10	28	28	5	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
+1288274400	2010	10	28	28	5	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
+1288274400	2010	10	28	28	5	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
+1288188000	2010	10	27	27	4	43	2010-10-27	2010-10-27	2010-10-29	2010-10-25	3952
+1288188000	2010	10	27	27	4	43	2010-10-27	2010-10-27	2010-10-29	2010-10-25	3952
+1288101600	2010	10	26	26	3	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
+1288101600	2010	10	26	26	3	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
+1288101600	2010	10	26	26	3	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
+1288101600	2010	10	26	26	3	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
+1288015200	2010	10	25	25	2	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
+1288015200	2010	10	25	25	2	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
+1288015200	2010	10	25	25	2	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
+1287928800	2010	10	24	24	1	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
+1287928800	2010	10	24	24	1	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
+1287928800	2010	10	24	24	1	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
+1287928800	2010	10	24	24	1	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
+1287842400	2010	10	23	23	7	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
+1287756000	2010	10	22	22	6	42	2010-10-22	2010-10-22	2010-10-24	2010-10-20	3947
+1287756000	2010	10	22	22	6	42	2010-10-22	2010-10-22	2010-10-24	2010-10-20	3947
+1287756000	2010	10	22	22	6	42	2010-10-22	2010-10-22	2010-10-24	2010-10-20	3947
+1287669600	2010	10	21	21	5	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
+1287669600	2010	10	21	21	5	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
+1287669600	2010	10	21	21	5	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
+1287583200	2010	10	20	20	4	42	2010-10-20	2010-10-20	2010-10-22	2010-10-18	3945
+1287583200	2010	10	20	20	4	42	2010-10-20	2010-10-20	2010-10-22	2010-10-18	3945
+1287842400	2010	10	23	23	7	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
+1287842400	2010	10	23	23	7	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
+1287842400	2010	10	23	23	7	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
+1288447200	2010	10	30	30	7	43	2010-10-30	2010-10-30	2010-11-01	2010-10-28	3955
+1288447200	2010	10	30	30	7	43	2010-10-30	2010-10-30	2010-11-01	2010-10-28	3955
+1287583200	2010	10	20	20	4	42	2010-10-20	2010-10-20	2010-10-22	2010-10-18	3945
+1287669600	2010	10	21	21	5	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
+1287842400	2010	10	23	23	7	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
+1287928800	2010	10	24	24	1	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
+1288015200	2010	10	25	25	2	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
+1288101600	2010	10	26	26	3	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
+1288188000	2010	10	27	27	4	43	2010-10-27	2010-10-27	2010-10-29	2010-10-25	3952
+1288274400	2010	10	28	28	5	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
+1288360800	2010	10	29	29	6	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
+1288533600	2010	10	31	31	1	43	2010-10-31	2010-10-31	2010-11-02	2010-10-29	3956
 PREHOOK: query: EXPLAIN SELECT
   to_unix_timestamp(fl_date),
   year(fl_date),
   month(fl_date),
   day(fl_date),
   dayofmonth(fl_date),
+  dayofweek(fl_date),
   weekofyear(fl_date),
   date(fl_date),
   to_date(fl_date),
@@ -446,6 +450,7 @@ POSTHOOK: query: EXPLAIN SELECT
   month(fl_date),
   day(fl_date),
   dayofmonth(fl_date),
+  dayofweek(fl_date),
   weekofyear(fl_date),
   date(fl_date),
   to_date(fl_date),
@@ -466,8 +471,8 @@ STAGE PLANS:
             alias: date_udf_flight_orc
             Statistics: Num rows: 137 Data size: 13152 Basic stats: COMPLETE Column stats: NONE
             Select Operator
-              expressions: to_unix_timestamp(fl_date) (type: bigint), year(fl_date) (type: int), month(fl_date) (type: int), day(fl_date) (type: int), dayofmonth(fl_date) (type: int), weekofyear(fl_date) (type: int), fl_date (type: date), to_date(fl_date) (type: string), date_add(fl_date, 2) (type: string), date_sub(fl_date, 2) (type: string), datediff(fl_date, '2000-01-01') (type: int)
-              outputColumnNames: _col0, _col1, _col2, _col3, _col4, _col5, _col6, _col7, _col8, _col9, _col10
+              expressions: to_unix_timestamp(fl_date) (type: bigint), year(fl_date) (type: int), month(fl_date) (type: int), day(fl_date) (type: int), dayofmonth(fl_date) (type: int), dayofweek(fl_date) (type: int), weekofyear(fl_date) (type: int), fl_date (type: date), to_date(fl_date) (type: string), date_add(fl_date, 2) (type: string), date_sub(fl_date, 2) (type: string), datediff(fl_date, '2000-01-01') (type: int)
+              outputColumnNames: _col0, _col1, _col2, _col3, _col4, _col5, _col6, _col7, _col8, _col9, _col10, _col11
               Statistics: Num rows: 137 Data size: 13152 Basic stats: COMPLETE Column stats: NONE
               File Output Operator
                 compressed: false
@@ -476,7 +481,6 @@ STAGE PLANS:
                     input format: org.apache.hadoop.mapred.TextInputFormat
                     output format: org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat
                     serde: org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe
-      Execution mode: vectorized
 
   Stage: Stage-0
     Fetch Operator
@@ -490,6 +494,7 @@ PREHOOK: query: SELECT
   month(fl_date),
   day(fl_date),
   dayofmonth(fl_date),
+  dayofweek(fl_date),
   weekofyear(fl_date),
   date(fl_date),
   to_date(fl_date),
@@ -506,6 +511,7 @@ POSTHOOK: query: SELECT
   month(fl_date),
   day(fl_date),
   dayofmonth(fl_date),
+  dayofweek(fl_date),
   weekofyear(fl_date),
   date(fl_date),
   to_date(fl_date),
@@ -516,148 +522,149 @@ FROM date_udf_flight_orc
 POSTHOOK: type: QUERY
 POSTHOOK: Input: default@date_udf_flight_orc
 #### A masked pattern was here ####
-1287558000	2010	10	20	20	42	2010-10-20	2010-10-20	2010-10-22	2010-10-18	3945
-1287558000	2010	10	20	20	42	2010-10-20	2010-10-20	2010-10-22	2010-10-18	3945
-1287558000	2010	10	20	20	42	2010-10-20	2010-10-20	2010-10-22	2010-10-18	3945
-1287558000	2010	10	20	20	42	2010-10-20	2010-10-20	2010-10-22	2010-10-18	3945
-1287558000	2010	10	20	20	42	2010-10-20	2010-10-20	2010-10-22	2010-10-18	3945
-1287558000	2010	10	20	20	42	2010-10-20	2010-10-20	2010-10-22	2010-10-18	3945
-1287558000	2010	10	20	20	42	2010-10-20	2010-10-20	2010-10-22	2010-10-18	3945
-1287558000	2010	10	20	20	42	2010-10-20	2010-10-20	2010-10-22	2010-10-18	3945
-1287644400	2010	10	21	21	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
-1287644400	2010	10	21	21	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
-1287644400	2010	10	21	21	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
-1287644400	2010	10	21	21	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
-1287644400	2010	10	21	21	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
-1287644400	2010	10	21	21	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
-1287644400	2010	10	21	21	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
-1287644400	2010	10	21	21	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
-1287730800	2010	10	22	22	42	2010-10-22	2010-10-22	2010-10-24	2010-10-20	3947
-1287730800	2010	10	22	22	42	2010-10-22	2010-10-22	2010-10-24	2010-10-20	3947
-1287730800	2010	10	22	22	42	2010-10-22	2010-10-22	2010-10-24	2010-10-20	3947
-1287730800	2010	10	22	22	42	2010-10-22	2010-10-22	2010-10-24	2010-10-20	3947
-1287730800	2010	10	22	22	42	2010-10-22	2010-10-22	2010-10-24	2010-10-20	3947
-1287730800	2010	10	22	22	42	2010-10-22	2010-10-22	2010-10-24	2010-10-20	3947
-1287730800	2010	10	22	22	42	2010-10-22	2010-10-22	2010-10-24	2010-10-20	3947
-1287730800	2010	10	22	22	42	2010-10-22	2010-10-22	2010-10-24	2010-10-20	3947
-1287817200	2010	10	23	23	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
-1287817200	2010	10	23	23	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
-1287817200	2010	10	23	23	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
-1287817200	2010	10	23	23	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
-1287817200	2010	10	23	23	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
-1287817200	2010	10	23	23	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
-1287817200	2010	10	23	23	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
-1287903600	2010	10	24	24	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
-1287903600	2010	10	24	24	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
-1287903600	2010	10	24	24	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
-1287903600	2010	10	24	24	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
-1287903600	2010	10	24	24	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
-1287903600	2010	10	24	24	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
-1287903600	2010	10	24	24	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
-1287990000	2010	10	25	25	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
-1287990000	2010	10	25	25	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
-1287990000	2010	10	25	25	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
-1287990000	2010	10	25	25	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
-1287990000	2010	10	25	25	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
-1287990000	2010	10	25	25	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
-1287990000	2010	10	25	25	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
-1287990000	2010	10	25	25	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
-1288076400	2010	10	26	26	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
-1288076400	2010	10	26	26	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
-1288076400	2010	10	26	26	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
-1288076400	2010	10	26	26	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
-1288076400	2010	10	26	26	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
-1288076400	2010	10	26	26	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
-1288076400	2010	10	26	26	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
-1288076400	2010	10	26	26	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
-1288162800	2010	10	27	27	43	2010-10-27	2010-10-27	2010-10-29	2010-10-25	3952
-1288162800	2010	10	27	27	43	2010-10-27	2010-10-27	2010-10-29	2010-10-25	3952
-1288162800	2010	10	27	27	43	2010-10-27	2010-10-27	2010-10-29	2010-10-25	3952
-1288162800	2010	10	27	27	43	2010-10-27	2010-10-27	2010-10-29	2010-10-25	3952
-1288162800	2010	10	27	27	43	2010-10-27	2010-10-27	2010-10-29	2010-10-25	3952
-1288162800	2010	10	27	27	43	2010-10-27	2010-10-27	2010-10-29	2010-10-25	3952
-1288162800	2010	10	27	27	43	2010-10-27	2010-10-27	2010-10-29	2010-10-25	3952
-1288162800	2010	10	27	27	43	2010-10-27	2010-10-27	2010-10-29	2010-10-25	3952
-1288249200	2010	10	28	28	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
-1288249200	2010	10	28	28	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
-1288249200	2010	10	28	28	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
-1288249200	2010	10	28	28	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
-1288249200	2010	10	28	28	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
-1288249200	2010	10	28	28	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
-1288249200	2010	10	28	28	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
-1288249200	2010	10	28	28	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
-1288335600	2010	10	29	29	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
-1288335600	2010	10	29	29	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
-1288335600	2010	10	29	29	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
-1288335600	2010	10	29	29	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
-1288335600	2010	10	29	29	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
-1288335600	2010	10	29	29	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
-1288335600	2010	10	29	29	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
-1288335600	2010	10	29	29	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
-1288422000	2010	10	30	30	43	2010-10-30	2010-10-30	2010-11-01	2010-10-28	3955
-1288422000	2010	10	30	30	43	2010-10-30	2010-10-30	2010-11-01	2010-10-28	3955
-1288422000	2010	10	30	30	43	2010-10-30	2010-10-30	2010-11-01	2010-10-28	3955
-1288422000	2010	10	30	30	43	2010-10-30	2010-10-30	2010-11-01	2010-10-28	3955
-1288422000	2010	10	30	30	43	2010-10-30	2010-10-30	2010-11-01	2010-10-28	3955
-1288422000	2010	10	30	30	43	2010-10-30	2010-10-30	2010-11-01	2010-10-28	3955
-1288422000	2010	10	30	30	43	2010-10-30	2010-10-30	2010-11-01	2010-10-28	3955
-1288508400	2010	10	31	31	43	2010-10-31	2010-10-31	2010-11-02	2010-10-29	3956
-1288508400	2010	10	31	31	43	2010-10-31	2010-10-31	2010-11-02	2010-10-29	3956
-1288508400	2010	10	31	31	43	2010-10-31	2010-10-31	2010-11-02	2010-10-29	3956
-1288508400	2010	10	31	31	43	2010-10-31	2010-10-31	2010-11-02	2010-10-29	3956
-1288508400	2010	10	31	31	43	2010-10-31	2010-10-31	2010-11-02	2010-10-29	3956
-1288508400	2010	10	31	31	43	2010-10-31	2010-10-31	2010-11-02	2010-10-29	3956
-1288508400	2010	10	31	31	43	2010-10-31	2010-10-31	2010-11-02	2010-10-29	3956
-1288422000	2010	10	30	30	43	2010-10-30	2010-10-30	2010-11-01	2010-10-28	3955
-1288422000	2010	10	30	30	43	2010-10-30	2010-10-30	2010-11-01	2010-10-28	3955
-1288335600	2010	10	29	29	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
-1288335600	2010	10	29	29	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
-1288335600	2010	10	29	29	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
-1288249200	2010	10	28	28	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
-1288249200	2010	10	28	28	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
-1288249200	2010	10	28	28	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
-1288162800	2010	10	27	27	43	2010-10-27	2010-10-27	2010-10-29	2010-10-25	3952
-1288162800	2010	10	27	27	43	2010-10-27	2010-10-27	2010-10-29	2010-10-25	3952
-1288076400	2010	10	26	26	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
-1288076400	2010	10	26	26	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
-1288076400	2010	10	26	26	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
-1288076400	2010	10	26	26	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
-1287990000	2010	10	25	25	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
-1287990000	2010	10	25	25	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
-1287990000	2010	10	25	25	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
-1287903600	2010	10	24	24	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
-1287903600	2010	10	24	24	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
-1287903600	2010	10	24	24	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
-1287903600	2010	10	24	24	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
-1287817200	2010	10	23	23	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
-1287730800	2010	10	22	22	42	2010-10-22	2010-10-22	2010-10-24	2010-10-20	3947
-1287730800	2010	10	22	22	42	2010-10-22	2010-10-22	2010-10-24	2010-10-20	3947
-1287730800	2010	10	22	22	42	2010-10-22	2010-10-22	2010-10-24	2010-10-20	3947
-1287644400	2010	10	21	21	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
-1287644400	2010	10	21	21	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
-1287644400	2010	10	21	21	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
-1287558000	2010	10	20	20	42	2010-10-20	2010-10-20	2010-10-22	2010-10-18	3945
-1287558000	2010	10	20	20	42	2010-10-20	2010-10-20	2010-10-22	2010-10-18	3945
-1287817200	2010	10	23	23	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
-1287817200	2010	10	23	23	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
-1287817200	2010	10	23	23	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
-1288422000	2010	10	30	30	43	2010-10-30	2010-10-30	2010-11-01	2010-10-28	3955
-1288422000	2010	10	30	30	43	2010-10-30	2010-10-30	2010-11-01	2010-10-28	3955
-1287558000	2010	10	20	20	42	2010-10-20	2010-10-20	2010-10-22	2010-10-18	3945
-1287644400	2010	10	21	21	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
-1287817200	2010	10	23	23	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
-1287903600	2010	10	24	24	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
-1287990000	2010	10	25	25	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
-1288076400	2010	10	26	26	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
-1288162800	2010	10	27	27	43	2010-10-27	2010-10-27	2010-10-29	2010-10-25	3952
-1288249200	2010	10	28	28	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
-1288335600	2010	10	29	29	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
-1288508400	2010	10	31	31	43	2010-10-31	2010-10-31	2010-11-02	2010-10-29	3956
+1287558000	2010	10	20	20	4	42	2010-10-20	2010-10-20	2010-10-22	2010-10-18	3945
+1287558000	2010	10	20	20	4	42	2010-10-20	2010-10-20	2010-10-22	2010-10-18	3945
+1287558000	2010	10	20	20	4	42	2010-10-20	2010-10-20	2010-10-22	2010-10-18	3945
+1287558000	2010	10	20	20	4	42	2010-10-20	2010-10-20	2010-10-22	2010-10-18	3945
+1287558000	2010	10	20	20	4	42	2010-10-20	2010-10-20	2010-10-22	2010-10-18	3945
+1287558000	2010	10	20	20	4	42	2010-10-20	2010-10-20	2010-10-22	2010-10-18	3945
+1287558000	2010	10	20	20	4	42	2010-10-20	2010-10-20	2010-10-22	2010-10-18	3945
+1287558000	2010	10	20	20	4	42	2010-10-20	2010-10-20	2010-10-22	2010-10-18	3945
+1287644400	2010	10	21	21	5	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
+1287644400	2010	10	21	21	5	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
+1287644400	2010	10	21	21	5	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
+1287644400	2010	10	21	21	5	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
+1287644400	2010	10	21	21	5	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
+1287644400	2010	10	21	21	5	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
+1287644400	2010	10	21	21	5	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
+1287644400	2010	10	21	21	5	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
+1287730800	2010	10	22	22	6	42	2010-10-22	2010-10-22	2010-10-24	2010-10-20	3947
+1287730800	2010	10	22	22	6	42	2010-10-22	2010-10-22	2010-10-24	2010-10-20	3947
+1287730800	2010	10	22	22	6	42	2010-10-22	2010-10-22	2010-10-24	2010-10-20	3947
+1287730800	2010	10	22	22	6	42	2010-10-22	2010-10-22	2010-10-24	2010-10-20	3947
+1287730800	2010	10	22	22	6	42	2010-10-22	2010-10-22	2010-10-24	2010-10-20	3947
+1287730800	2010	10	22	22	6	42	2010-10-22	2010-10-22	2010-10-24	2010-10-20	3947
+1287730800	2010	10	22	22	6	42	2010-10-22	2010-10-22	2010-10-24	2010-10-20	3947
+1287730800	2010	10	22	22	6	42	2010-10-22	2010-10-22	2010-10-24	2010-10-20	3947
+1287817200	2010	10	23	23	7	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
+1287817200	2010	10	23	23	7	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
+1287817200	2010	10	23	23	7	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
+1287817200	2010	10	23	23	7	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
+1287817200	2010	10	23	23	7	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
+1287817200	2010	10	23	23	7	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
+1287817200	2010	10	23	23	7	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
+1287903600	2010	10	24	24	1	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
+1287903600	2010	10	24	24	1	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
+1287903600	2010	10	24	24	1	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
+1287903600	2010	10	24	24	1	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
+1287903600	2010	10	24	24	1	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
+1287903600	2010	10	24	24	1	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
+1287903600	2010	10	24	24	1	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
+1287990000	2010	10	25	25	2	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
+1287990000	2010	10	25	25	2	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
+1287990000	2010	10	25	25	2	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
+1287990000	2010	10	25	25	2	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
+1287990000	2010	10	25	25	2	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
+1287990000	2010	10	25	25	2	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
+1287990000	2010	10	25	25	2	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
+1287990000	2010	10	25	25	2	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
+1288076400	2010	10	26	26	3	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
+1288076400	2010	10	26	26	3	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
+1288076400	2010	10	26	26	3	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
+1288076400	2010	10	26	26	3	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
+1288076400	2010	10	26	26	3	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
+1288076400	2010	10	26	26	3	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
+1288076400	2010	10	26	26	3	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
+1288076400	2010	10	26	26	3	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
+1288162800	2010	10	27	27	4	43	2010-10-27	2010-10-27	2010-10-29	2010-10-25	3952
+1288162800	2010	10	27	27	4	43	2010-10-27	2010-10-27	2010-10-29	2010-10-25	3952
+1288162800	2010	10	27	27	4	43	2010-10-27	2010-10-27	2010-10-29	2010-10-25	3952
+1288162800	2010	10	27	27	4	43	2010-10-27	2010-10-27	2010-10-29	2010-10-25	3952
+1288162800	2010	10	27	27	4	43	2010-10-27	2010-10-27	2010-10-29	2010-10-25	3952
+1288162800	2010	10	27	27	4	43	2010-10-27	2010-10-27	2010-10-29	2010-10-25	3952
+1288162800	2010	10	27	27	4	43	2010-10-27	2010-10-27	2010-10-29	2010-10-25	3952
+1288162800	2010	10	27	27	4	43	2010-10-27	2010-10-27	2010-10-29	2010-10-25	3952
+1288249200	2010	10	28	28	5	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
+1288249200	2010	10	28	28	5	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
+1288249200	2010	10	28	28	5	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
+1288249200	2010	10	28	28	5	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
+1288249200	2010	10	28	28	5	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
+1288249200	2010	10	28	28	5	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
+1288249200	2010	10	28	28	5	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
+1288249200	2010	10	28	28	5	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
+1288335600	2010	10	29	29	6	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
+1288335600	2010	10	29	29	6	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
+1288335600	2010	10	29	29	6	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
+1288335600	2010	10	29	29	6	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
+1288335600	2010	10	29	29	6	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
+1288335600	2010	10	29	29	6	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
+1288335600	2010	10	29	29	6	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
+1288335600	2010	10	29	29	6	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
+1288422000	2010	10	30	30	7	43	2010-10-30	2010-10-30	2010-11-01	2010-10-28	3955
+1288422000	2010	10	30	30	7	43	2010-10-30	2010-10-30	2010-11-01	2010-10-28	3955
+1288422000	2010	10	30	30	7	43	2010-10-30	2010-10-30	2010-11-01	2010-10-28	3955
+1288422000	2010	10	30	30	7	43	2010-10-30	2010-10-30	2010-11-01	2010-10-28	3955
+1288422000	2010	10	30	30	7	43	2010-10-30	2010-10-30	2010-11-01	2010-10-28	3955
+1288422000	2010	10	30	30	7	43	2010-10-30	2010-10-30	2010-11-01	2010-10-28	3955
+1288422000	2010	10	30	30	7	43	2010-10-30	2010-10-30	2010-11-01	2010-10-28	3955
+1288508400	2010	10	31	31	1	43	2010-10-31	2010-10-31	2010-11-02	2010-10-29	3956
+1288508400	2010	10	31	31	1	43	2010-10-31	2010-10-31	2010-11-02	2010-10-29	3956
+1288508400	2010	10	31	31	1	43	2010-10-31	2010-10-31	2010-11-02	2010-10-29	3956
+1288508400	2010	10	31	31	1	43	2010-10-31	2010-10-31	2010-11-02	2010-10-29	3956
+1288508400	2010	10	31	31	1	43	2010-10-31	2010-10-31	2010-11-02	2010-10-29	3956
+1288508400	2010	10	31	31	1	43	2010-10-31	2010-10-31	2010-11-02	2010-10-29	3956
+1288508400	2010	10	31	31	1	43	2010-10-31	2010-10-31	2010-11-02	2010-10-29	3956
+1288422000	2010	10	30	30	7	43	2010-10-30	2010-10-30	2010-11-01	2010-10-28	3955
+1288422000	2010	10	30	30	7	43	2010-10-30	2010-10-30	2010-11-01	2010-10-28	3955
+1288335600	2010	10	29	29	6	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
+1288335600	2010	10	29	29	6	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
+1288335600	2010	10	29	29	6	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
+1288249200	2010	10	28	28	5	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
+1288249200	2010	10	28	28	5	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
+1288249200	2010	10	28	28	5	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
+1288162800	2010	10	27	27	4	43	2010-10-27	2010-10-27	2010-10-29	2010-10-25	3952
+1288162800	2010	10	27	27	4	43	2010-10-27	2010-10-27	2010-10-29	2010-10-25	3952
+1288076400	2010	10	26	26	3	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
+1288076400	2010	10	26	26	3	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
+1288076400	2010	10	26	26	3	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
+1288076400	2010	10	26	26	3	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
+1287990000	2010	10	25	25	2	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
+1287990000	2010	10	25	25	2	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
+1287990000	2010	10	25	25	2	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
+1287903600	2010	10	24	24	1	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
+1287903600	2010	10	24	24	1	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
+1287903600	2010	10	24	24	1	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
+1287903600	2010	10	24	24	1	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
+1287817200	2010	10	23	23	7	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
+1287730800	2010	10	22	22	6	42	2010-10-22	2010-10-22	2010-10-24	2010-10-20	3947
+1287730800	2010	10	22	22	6	42	2010-10-22	2010-10-22	2010-10-24	2010-10-20	3947
+1287730800	2010	10	22	22	6	42	2010-10-22	2010-10-22	2010-10-24	2010-10-20	3947
+1287644400	2010	10	21	21	5	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
+1287644400	2010	10	21	21	5	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
+1287644400	2010	10	21	21	5	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
+1287558000	2010	10	20	20	4	42	2010-10-20	2010-10-20	2010-10-22	2010-10-18	3945
+1287558000	2010	10	20	20	4	42	2010-10-20	2010-10-20	2010-10-22	2010-10-18	3945
+1287817200	2010	10	23	23	7	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
+1287817200	2010	10	23	23	7	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
+1287817200	2010	10	23	23	7	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
+1288422000	2010	10	30	30	7	43	2010-10-30	2010-10-30	2010-11-01	2010-10-28	3955
+1288422000	2010	10	30	30	7	43	2010-10-30	2010-10-30	2010-11-01	2010-10-28	3955
+1287558000	2010	10	20	20	4	42	2010-10-20	2010-10-20	2010-10-22	2010-10-18	3945
+1287644400	2010	10	21	21	5	42	2010-10-21	2010-10-21	2010-10-23	2010-10-19	3946
+1287817200	2010	10	23	23	7	42	2010-10-23	2010-10-23	2010-10-25	2010-10-21	3948
+1287903600	2010	10	24	24	1	42	2010-10-24	2010-10-24	2010-10-26	2010-10-22	3949
+1287990000	2010	10	25	25	2	43	2010-10-25	2010-10-25	2010-10-27	2010-10-23	3950
+1288076400	2010	10	26	26	3	43	2010-10-26	2010-10-26	2010-10-28	2010-10-24	3951
+1288162800	2010	10	27	27	4	43	2010-10-27	2010-10-27	2010-10-29	2010-10-25	3952
+1288249200	2010	10	28	28	5	43	2010-10-28	2010-10-28	2010-10-30	2010-10-26	3953
+1288335600	2010	10	29	29	6	43	2010-10-29	2010-10-29	2010-10-31	2010-10-27	3954
+1288508400	2010	10	31	31	1	43	2010-10-31	2010-10-31	2010-11-02	2010-10-29	3956
 PREHOOK: query: EXPLAIN SELECT
   year(fl_time) = year(fl_date),
   month(fl_time) = month(fl_date),
   day(fl_time) = day(fl_date),
   dayofmonth(fl_time) = dayofmonth(fl_date),
+  dayofweek(fl_time) = dayofweek(fl_date),
   weekofyear(fl_time) = weekofyear(fl_date),
   date(fl_time) = date(fl_date),
   to_date(fl_time) = to_date(fl_date),
@@ -671,6 +678,7 @@ POSTHOOK: query: EXPLAIN SELECT
   month(fl_time) = month(fl_date),
   day(fl_time) = day(fl_date),
   dayofmonth(fl_time) = dayofmonth(fl_date),
+  dayofweek(fl_time) = dayofweek(fl_date),
   weekofyear(fl_time) = weekofyear(fl_date),
   date(fl_time) = date(fl_date),
   to_date(fl_time) = to_date(fl_date),
@@ -691,8 +699,8 @@ STAGE PLANS:
             alias: date_udf_flight_orc
             Statistics: Num rows: 137 Data size: 13152 Basic stats: COMPLETE Column stats: NONE
             Select Operator
-              expressions: (year(fl_time) = year(fl_date)) (type: boolean), (month(fl_time) = month(fl_date)) (type: boolean), (day(fl_time) = day(fl_date)) (type: boolean), (dayofmonth(fl_time) = dayofmonth(fl_date)) (type: boolean), (weekofyear(fl_time) = weekofyear(fl_date)) (type: boolean), (CAST( fl_time AS DATE) = fl_date) (type: boolean), (to_date(fl_time) = to_date(fl_date)) (type: boolean), (date_add(fl_time, 2) = date_add(fl_date, 2)) (type: boolean), (date_sub(fl_time, 2) = date_sub(fl_date, 2)) (type: boolean), (datediff(fl_time, '2000-01-01') = datediff(fl_date, '2000-01-01')) (type: boolean)
-              outputColumnNames: _col0, _col1, _col2, _col3, _col4, _col5, _col6, _col7, _col8, _col9
+              expressions: (year(fl_time) = year(fl_date)) (type: boolean), (month(fl_time) = month(fl_date)) (type: boolean), (day(fl_time) = day(fl_date)) (type: boolean), (dayofmonth(fl_time) = dayofmonth(fl_date)) (type: boolean), (dayofweek(fl_time) = dayofweek(fl_date)) (type: boolean), (weekofyear(fl_time) = weekofyear(fl_date)) (type: boolean), (CAST( fl_time AS DATE) = fl_date) (type: boolean), (to_date(fl_time) = to_date(fl_date)) (type: boolean), (date_add(fl_time, 2) = date_add(fl_date, 2)) (type: boolean), (date_sub(fl_time, 2) = date_sub(fl_date, 2)) (type: boolean), (datediff(fl_time, '2000-01-01') = datediff(fl_date, '2000-01-01')) (type: boolean)
+              outputColumnNames: _col0, _col1, _col2, _col3, _col4, _col5, _col6, _col7, _col8, _col9, _col10
               Statistics: Num rows: 137 Data size: 13152 Basic stats: COMPLETE Column stats: NONE
               File Output Operator
                 compressed: false
@@ -701,7 +709,6 @@ STAGE PLANS:
                     input format: org.apache.hadoop.mapred.TextInputFormat
                     output format: org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat
                     serde: org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe
-      Execution mode: vectorized
 
   Stage: Stage-0
     Fetch Operator
@@ -715,6 +722,7 @@ SELECT
   month(fl_time) = month(fl_date),
   day(fl_time) = day(fl_date),
   dayofmonth(fl_time) = dayofmonth(fl_date),
+  dayofweek(fl_time) = dayofweek(fl_date),
   weekofyear(fl_time) = weekofyear(fl_date),
   date(fl_time) = date(fl_date),
   to_date(fl_time) = to_date(fl_date),
@@ -731,6 +739,7 @@ SELECT
   month(fl_time) = month(fl_date),
   day(fl_time) = day(fl_date),
   dayofmonth(fl_time) = dayofmonth(fl_date),
+  dayofweek(fl_time) = dayofweek(fl_date),
   weekofyear(fl_time) = weekofyear(fl_date),
   date(fl_time) = date(fl_date),
   to_date(fl_time) = to_date(fl_date),
@@ -741,143 +750,143 @@ FROM date_udf_flight_orc
 POSTHOOK: type: QUERY
 POSTHOOK: Input: default@date_udf_flight_orc
 #### A masked pattern was here ####
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
-true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
+true	true	true	true	true	true	true	true	true	true	true
 PREHOOK: query: EXPLAIN SELECT 
   fl_date, 
   to_date(date_add(fl_date, 2)), 
-- 
1.7.9.5

